<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from 4leg_robot.xacro               | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="4leg_robot" xmlns:xacro="http://ros.org/wiki/xacro">
  <material name="gray">
    <color rgba="0.5 0.5 0.5 2.5"/>
  </material>
  <link name="base_link"/>
  <joint name="robot_base" type="fixed">
    <parent link="base_link"/>
    <child link="robot_body"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0101"/>
  </joint>
  <link name="robot_body">
    <visual>
      <geometry>
        <mesh filename="package://4leg_robot/stl/Servo_4leg_body.stl"/>
      </geometry>
      <material name="gray"/>
    </visual>
  </link>
  <joint name="right_leg_front_joint" type="continuous">
    <origin rpy="0 0 0" xyz="-0.141 -0.0572 0.0"/>
    <parent link="base_link"/>
    <child link="right_leg_front_servohead5"/>
    <axis xyz="1 0 0"/>
  </joint>
  <link name="right_leg_front_servohead5">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://4leg_robot/stl/servohead_parts5.stl"/>
      </geometry>
      <material name="gray"/>
    </visual>
  </link>
  <joint name="right_leg_front_servohead_joint" type="fixed">
    <parent link="right_leg_front_servohead5"/>
    <child link="right_leg_front_servohead6"/>
    <origin xyz="0.0 0.015 0.0"/>
  </joint>
  <link name="right_leg_front_servohead6">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://4leg_robot/stl/servohead_parts6.stl"/>
      </geometry>
      <material name="gray"/>
    </visual>
  </link>
  <joint name="right_leg_front_leg_joint1" type="continuous">
    <parent link="right_leg_front_servohead5"/>
    <child link="right_leg_front_leg1"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 -0.016 0.0"/>
    <axis xyz="0 1 0"/>
  </joint>
  <link name="right_leg_front_leg1">
    <visual>
      <geometry>
        <mesh filename="package://4leg_robot/stl/leg_parts1.stl"/>
      </geometry>
      <material name="gray"/>
    </visual>
  </link>
  <joint name="right_leg_front_leg_joint2" type="continuous">
    <parent link="right_leg_front_leg1"/>
    <child link="right_leg_front_leg2"/>
    <origin rpy="0.0 1.57079632679 0.0" xyz="0.13 -0.019 0.0"/>
    <axis xyz="0 1 0"/>
  </joint>
  <link name="right_leg_front_leg2">
    <visual>
      <geometry>
        <mesh filename="package://4leg_robot/stl/leg_parts2.stl"/>
      </geometry>
      <material name="gray"/>
    </visual>
  </link>
  <joint name="right_leg_front_leg_position" type="fixed">
    <parent link="right_leg_front_leg2"/>
    <child link="right_leg_front_toe"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.127 -0.005 0.0"/>
  </joint>
  <link name="right_leg_front_toe">
    <visual>
      <geometry>
        <sphere radius="0.001"/>
      </geometry>
      <material name="gray"/>
    </visual>
  </link>
  <joint name="left_leg_front_joint" type="continuous">
    <origin rpy="0 0 0" xyz="0.141 -0.0572 0.0"/>
    <parent link="base_link"/>
    <child link="left_leg_front_servohead7"/>
    <axis xyz="1 0 0"/>
  </joint>
  <link name="left_leg_front_servohead7">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://4leg_robot/stl/servohead_parts7.stl"/>
      </geometry>
      <material name="gray"/>
    </visual>
  </link>
  <joint name="left_leg_front_servohead_joint" type="fixed">
    <parent link="left_leg_front_servohead7"/>
    <child link="left_leg_front_servohead6"/>
    <origin xyz="0.0 0.015 0.0"/>
  </joint>
  <link name="left_leg_front_servohead6">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://4leg_robot/stl/servohead_parts6.stl"/>
      </geometry>
      <material name="gray"/>
    </visual>
  </link>
  <joint name="left_leg_front_leg_joint1" type="continuous">
    <parent link="left_leg_front_servohead7"/>
    <child link="left_leg_front_leg1"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 -0.016 0.0"/>
    <axis xyz="0 1 0"/>
  </joint>
  <link name="left_leg_front_leg1">
    <visual>
      <geometry>
        <mesh filename="package://4leg_robot/stl/leg_parts1.stl"/>
      </geometry>
      <material name="gray"/>
    </visual>
  </link>
  <joint name="left_leg_front_leg_joint2" type="continuous">
    <parent link="left_leg_front_leg1"/>
    <child link="left_leg_front_leg2"/>
    <origin rpy="0.0 1.57079632679 0.0" xyz="0.13 -0.019 0.0"/>
    <axis xyz="0 1 0"/>
  </joint>
  <link name="left_leg_front_leg2">
    <visual>
      <geometry>
        <mesh filename="package://4leg_robot/stl/leg_parts3.stl"/>
      </geometry>
      <material name="gray"/>
    </visual>
  </link>
  <joint name="left_leg_front_leg_position" type="fixed">
    <parent link="left_leg_front_leg2"/>
    <child link="left_leg_front_toe"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.127 -0.005 0.0"/>
  </joint>
  <link name="left_leg_front_toe">
    <visual>
      <geometry>
        <sphere radius="0.001"/>
      </geometry>
      <material name="gray"/>
    </visual>
  </link>
  <joint name="right_leg_back_joint" type="continuous">
    <origin rpy="0 0 -3.14159265359" xyz="-0.141 0.0572 0.0"/>
    <parent link="base_link"/>
    <child link="right_leg_back_servohead7"/>
    <axis xyz="1 0 0"/>
  </joint>
  <link name="right_leg_back_servohead7">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://4leg_robot/stl/servohead_parts7.stl"/>
      </geometry>
      <material name="gray"/>
    </visual>
  </link>
  <joint name="right_leg_back_servohead_joint" type="fixed">
    <parent link="right_leg_back_servohead7"/>
    <child link="right_leg_back_servohead6"/>
    <origin xyz="0.0 0.015 0.0"/>
  </joint>
  <link name="right_leg_back_servohead6">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://4leg_robot/stl/servohead_parts6.stl"/>
      </geometry>
      <material name="gray"/>
    </visual>
  </link>
  <joint name="right_leg_back_leg_joint1" type="continuous">
    <parent link="right_leg_back_servohead7"/>
    <child link="right_leg_back_leg1"/>
    <origin rpy="0.0 3.14159265359 0.0" xyz="0.0 -0.016 0.0"/>
    <axis xyz="0 1 0"/>
  </joint>
  <link name="right_leg_back_leg1">
    <visual>
      <geometry>
        <mesh filename="package://4leg_robot/stl/leg_parts1.stl"/>
      </geometry>
      <material name="gray"/>
    </visual>
  </link>
  <joint name="right_leg_back_leg_joint2" type="continuous">
    <parent link="right_leg_back_leg1"/>
    <child link="right_leg_back_leg2"/>
    <origin rpy="0.0 -1.57079632679 0.0" xyz="0.13 -0.019 0.0"/>
    <axis xyz="0 1 0"/>
  </joint>
  <link name="right_leg_back_leg2">
    <visual>
      <geometry>
        <mesh filename="package://4leg_robot/stl/leg_parts2.stl"/>
      </geometry>
      <material name="gray"/>
    </visual>
  </link>
  <joint name="right_leg_back_leg_position" type="fixed">
    <parent link="right_leg_back_leg2"/>
    <child link="right_leg_back_toe"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.127 -0.005 0.0"/>
  </joint>
  <link name="right_leg_back_toe">
    <visual>
      <geometry>
        <sphere radius="0.001"/>
      </geometry>
      <material name="gray"/>
    </visual>
  </link>
  <joint name="left_leg_back_joint" type="continuous">
    <origin rpy="0 0 -3.14159265359" xyz="0.141 0.0572 0.0"/>
    <parent link="base_link"/>
    <child link="left_leg_back_servohead5"/>
    <axis xyz="1 0 0"/>
  </joint>
  <link name="left_leg_back_servohead5">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://4leg_robot/stl/servohead_parts5.stl"/>
      </geometry>
      <material name="gray"/>
    </visual>
  </link>
  <joint name="left_leg_back_servohead_joint" type="fixed">
    <parent link="left_leg_back_servohead5"/>
    <child link="left_leg_back_servohead6"/>
    <origin xyz="0.0 0.015 0.0"/>
  </joint>
  <link name="left_leg_back_servohead6">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://4leg_robot/stl/servohead_parts6.stl"/>
      </geometry>
      <material name="gray"/>
    </visual>
  </link>
  <joint name="left_leg_back_leg_joint1" type="continuous">
    <parent link="left_leg_back_servohead5"/>
    <child link="left_leg_back_leg1"/>
    <origin rpy="0.0 3.14159265359 0.0" xyz="0.0 -0.016 0.0"/>
    <axis xyz="0 1 0"/>
  </joint>
  <link name="left_leg_back_leg1">
    <visual>
      <geometry>
        <mesh filename="package://4leg_robot/stl/leg_parts1.stl"/>
      </geometry>
      <material name="gray"/>
    </visual>
  </link>
  <joint name="left_leg_back_leg_joint2" type="continuous">
    <parent link="left_leg_back_leg1"/>
    <child link="left_leg_back_leg2"/>
    <origin rpy="0.0 -1.57079632679 0.0" xyz="0.13 -0.019 0.0"/>
    <axis xyz="0 1 0"/>
  </joint>
  <link name="left_leg_back_leg2">
    <visual>
      <geometry>
        <mesh filename="package://4leg_robot/stl/leg_parts3.stl"/>
      </geometry>
      <material name="gray"/>
    </visual>
  </link>
  <joint name="left_leg_back_leg_position" type="fixed">
    <parent link="left_leg_back_leg2"/>
    <child link="left_leg_back_toe"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.127 -0.005 0.0"/>
  </joint>
  <link name="left_leg_back_toe">
    <visual>
      <geometry>
        <sphere radius="0.001"/>
      </geometry>
      <material name="gray"/>
    </visual>
  </link>
</robot>

